test/integration/api-create.test.js

```diff
@@ -90,250 +93,481 @@ test('setup', function (tt) {
+//         h.createDockerContainer({
+//             vmapiClient: VMAPI,
+//             dockerClient: DOCKER_ALICE,
+//             test: t,
+//             extra: { 'Env': ['ENV_NO_VALUE'] },
+//             start: true  // Will start the container after creating.
+//         }, oncreate);
+
+//         function oncreate(err, result) {
+//             t.ifErr(err, 'create empty-env-var container');
+//             t.equal(result.vm.state, 'running', 'Check container running');
+//             DOCKER_ALICE.del('/containers/' + result.id + '?force=1', ondelete);
+//         }
+
+//         function ondelete(err) {
+//             t.ifErr(err, 'delete empty-env-var container');
+//             t.end();
+//         }
+//     });
+// });
+
+test('ensure fabrics enabled', function (tt) {
+    tt.test('fabric configuration', function (t) {
+        var listOpts = {};
+        var listParams = {};
+        NAPI.listFabricVLANs(ALICE.account.uuid, listOpts, listParams,
+            function (err, vlans) {
+                if (err) {
+                    FABRICS = false;
+                    if (err.code !== 'PreconditionRequiredError') {
+                        t.ifErr(err);
+                    }
+                } else {
+                    FABRICS = true;
+                }
+                t.end();
         });
+    });
 });
 
 
-test('api: create', function (tt) {
-
-    var created;
-
-    tt.test('docker create', function (t) {
-        h.createDockerContainer({
-            vmapiClient: VMAPI,
-            dockerClient: DOCKER_ALICE,
-            test: t
-        }, oncreate);
-
-        function oncreate(err, result) {
-            t.ifErr(err, 'create container');
-            created = result;
-            t.end();
-        }
+/*
+ * Creates two fabric networks, with the second named using the uuid of the
+ * first, used to test HostConfig.NetworkMode handling.
+ */
+test('fabric setup', function (tt) {
+    // set fabric status
+
+    var fVlan;
+    var fNetwork1;
+    var fNetwork2;
+
+    if (!FABRICS) {
+        tt.end();
+        return;
+    }
+
+    tt.test('fabric vlan setup', function (t) {
+        // create a new one.
+        var fabricParams = {
+            name: 'alicetest',
+            description: 'integration test fixture',
+            vlan_id: 4
+        };
+        h.getOrCreateFabricVLAN(NAPI, ALICE.account.uuid, fabricParams,
+            function (err, vlan) {
+                t.ifErr(err, 'create fabric vlan');
+                fVlan = vlan;
+                t.end();
+                return;
+            }
+        );
     });
 
-    tt.test('docker rm', function (t) {
-        DOCKER_ALICE.del('/containers/' + created.id, ondel);
-
-        function ondel(err, res, req, body) {
-            t.ifErr(err, 'rm container');
+    tt.test('fabric network setup', function (t) {
+        var nw1uuid = libuuid.create();
+        var nw2uuid = nw1uuid.replace(/-/g, '').substr(0, 12);
```

Comment:
Would it be worth it to add a comment here that the UUID of the second network is meant to be created in a way that exercises the code that disambiguate network IDs, short IDs and names?